lib/auth/db.go

```diff
@@ -89,7 +99,7 @@ func getServerNames(req *proto.DatabaseCertRequest) []string {
 
 // SignDatabaseCSR generates a client certificate used by proxy when talking
 // to a remote database service.
-func (s *Server) SignDatabaseCSR(ctx context.Context, req *proto.DatabaseCSRRequest) (*proto.DatabaseCSRResponse, error) {
+func (s *Server) SignDatabaseCSR(_ context.Context, req *proto.DatabaseCSRRequest) (*proto.DatabaseCSRResponse, error) {
```

Comment:
@jakule I have just noticed that we're still using HostCA here. I think it should be updated as well.

Reply:
@r0mant 
The HostCA flag in scope of  SignDatabaseCSR method is used to obtain HostCA certificate and add HostCA to the trusted pool were will used during reverse tunnel connection [teleport db proxy <--> db service ] mTLS handshake. DB service terminates the incoming connection with a server identity that was generated by HostCA during RegisterUsingToken (or LocalRegister if DB service is running with auth service) call. 

The connection established with a cert generated by the `SignDatabaseCSR` method is teleport "internal component" connection between teleport proxy and DB agent thus do we really want to introduce different server identity generation for `RoleDatabase` SystemRole for DB service and sign RoleDatabase identity with DatabaseCA ?

Reply:
@smallinsky @jakule So there are two aspects to it I think:

- There's an agent (process) identity that gets generated when a new agent starts up. Currently it is signed by the Host CA. I think that ideally each agent's identity would also be signed by their respective CA (i.e. SSH nodes - by Host CA, database agents - by Database CA, desktop agents - by Desktop CA, etc.) but that would likely be a pretty large change and it's a little orthogonal to what this PR does. So I propose to leave agent identity as it is for now.

- There's a certificate that's generated by this SignDatabaseCSR method that as you guys mentioned is used to transfer identity information over reverse tunnel to the database agent. I think it should use Database CA instead of Host CA as I suggested here. It makes sense that when communicating with a database agent the proxy would use the Database CA to sign those certs. This way one compromised CA does not affect others. It should be pretty easy to do if we make sure that database agent loads both Host CA and Database CA in its client cert pool.

One thing to keep in mind is that we should maintain backwards compatibility to avoid situation where a new Proxy signs a certificate with Database CA which older database agent can't verify (our upgrade instructions say to upgrade Proxies before agents). To make sure compatibility is not broken, Proxy can check the version of the agent it's connecting to (Teleport version should be available as a part of DatabaseServer resource) and sign the cert with Host CA if it's talking to an older agent (e.g. < 9.x) and with Database CA otherwise. Then in Teleport 10 we will remove Host CA from database agents' client pools.

WDYGT?

Reply:
Yes, Agree Connecting to an DB agent service with certs generated by DatabaseCA based on the agent version where DatabaseCA will be added to the  DB agent poll sounds reasonable and should be pretty easy to implement. 
Though without changing agent (process) identity signer to `DatabaseCA` we will still be using `HostCA` pool on the proxy side to verify cert presented by DB agent during reverse tunnel connection. But changing this might be more complicated ( for example identify is generated once and cached on local agent backend) and probably should be kept out of scope for this PR.